plugins/file-open-save-new/core/src/test/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProviderTest.java

```diff
@@ -0,0 +1,215 @@
+    ConnectionDetails mockConnectionDetails  = mock( ConnectionDetails.class );
+    assertFalse( testInstance.hasBuckets( mockConnectionDetails ) );
+
+    // TEST  does not have buckets
+    VFSConnectionDetails mock_NoBuckets_VFSConnectionDetails  = mock( VFSConnectionDetails.class );
+    when( mock_NoBuckets_VFSConnectionDetails.hasBuckets() ).thenReturn( false );
+    assertFalse( testInstance.hasBuckets( mock_NoBuckets_VFSConnectionDetails ) );
+
+    // TEST has buckets
+    VFSConnectionDetails mock_Buckets_VFSConnectionDetails  = mock( VFSConnectionDetails.class );
+    when( mock_Buckets_VFSConnectionDetails.hasBuckets() ).thenReturn( true );
+    assertTrue( testInstance.hasBuckets( mock_Buckets_VFSConnectionDetails ) );
+  }
+
+  @Test
+  public void testIsSupported() throws Exception {
+
+    VFSFileProvider testInstance = createTestInstance();
+
+    assertFalse( testInstance.isSupported( null ) );
+
+    assertFalse( testInstance.isSupported( "" ) );
+
+    assertFalse( testInstance.isSupported( "    " ) );
+
+    assertFalse( testInstance.isSupported( "someGarbage" ) );
+
+    assertFalse( testInstance.isSupported( "/someUser/someUnixFile" ) );
+
+    assertFalse( testInstance.isSupported( "T:\\Users\\RandomSUser\\Documents\\someWindowsFile" ) );
+
+    assertFalse( testInstance.isSupported( "//home/randomUser/randomFile.rpt" ) );
+
+    assertTrue( testInstance.isSupported( "pvfs://someConenction/someFilePath" ) );
+  }
+
+  @Test
+  public void testGetConnectionName() throws Exception {
+
+    VFSFileProvider testInstance = createTestInstance();
+
+    assertNull( testInstance.getConnectionName( createTestInstance( null ) ) );
+
+    assertNull( testInstance.getConnectionName( createTestInstance( "" ) ) );
+
+    assertNull( testInstance.getConnectionName( createTestInstance( "    " ) ) );
+
+    assertNull( testInstance.getConnectionName( createTestInstance( "someGarbage" ) ) );
+
+    assertNull( testInstance.getConnectionName( createTestInstance( "xyz:/123" ) ) ); // missing slash "/"
+
+    assertNull( testInstance.getConnectionName( createTestInstance( "xyz://" ) ) );
+
+    assertEquals("abc", testInstance.getConnectionName( createTestInstance( "xyz://abc" ) ) );
+
+    assertEquals("abc", testInstance.getConnectionName( createTestInstance( "xyz://abc/" ) ) );
+  }
+
+  @Test
+  public void testGetScheme() throws Exception {
+
+    VFSFileProvider testInstance = createTestInstance();
+
+    assertNull( testInstance.getScheme( createTestInstance( null ) ) );
+
+    assertNull( testInstance.getScheme( createTestInstance( "" ) ) );
+
+    assertNull( testInstance.getScheme( createTestInstance( "    " ) ) );
+
+    assertNull( testInstance.getScheme( createTestInstance( "someGarbage" ) ) );
+
+    assertEquals( "xyz", testInstance.getScheme( createTestInstance( "xyz://abc" ) ) );
+  }
+
+  @Test
+  public void testIsSame_DifferentTypes() throws Exception {
+    VFSFileProvider testInstance = createTestInstance();
```

Comment:
Should include tests with connection names with special characters.

Reply:
Addressed see commit https://github.com/pentaho/pentaho-kettle/commit/3e8bcc9f08d5634a303f5bdcbaf73f80d1811e46